src/policy/protect_payload.rs

```diff
@@ -103,78 +81,53 @@ impl PolicyModule for ProtectPayloadPolicy {
             }
         }
 
-        // Step 2: Save sensitives privileged registers
-        self.confidential_values_payload = get_confidential_values(ctx);
-        self.write_confidential_registers(ctx, false);
-
-        // Step 3: Lock memory
+        // Lock memory
         mctx.pmp
-            .set_from_policy(0, build_napot(0x80400000, 0x80000).unwrap(), pmpcfg::NAPOT);
+            .set_from_policy(0, 0x80400000 / 4, pmpcfg::INACTIVE);
+        mctx.pmp.set_from_policy(1, usize::MAX / 4, pmpcfg::TOR);
+
+        self.last_cause = ctx.trap_info.get_cause();
+
+        /*if !self.protected {
+            return;
+        } else {
+            // todo: In the future, add CSR handling here
+            return;
+        }*/
```

Comment:
Any reason to keep that? If yet let's add a comment to explain why, otherwise let's get rid of it.

Reply:
Well if you need it for the next commit I suggest you start working on the next commit on this branch and we merge everything once it is working together :) 

Committing intermediate state is not great in general, it might get lost and stay there forever for instance. It's fine to work on a few commits at a time on the same branch and refine them over time.